### PR TITLE
1953 joi env validation

### LINKED ISSUE #1953: Create Joi validation for env variables in all services
- [ ] The feature was discussed in [discussions](https://github.com/openkfw/TruBudget/discussions) of this repository.
- [ ] I have searched the [issues](https://github.com/openkfw/TruBudget/issues) of this repository and believe that this is not a duplicate.

## Description 💡

- validate env variables with Joi validation
- add script for generating markdown file `environment-variables.md` in each service
- add running of script in all services into release procedure
- add a separate script that will allow to run validation of present env variables against the validation without starting of service

Discussion-reference: <Insert discussion reference here>

### PR DESCRIPTION
**Pull request by galethil**
### Checklist

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/openkfw/TruBudget/blob/main/.github/CONTRIBUTING.md#open-a-pull-request).
- [x] I fixed all necessary PR warnings
- [x] The commit history is clean
- [ ] The E2E tests are passing
- [x] If possible, the issue has been divided into more subtasks
- [x] I did a self review before requesting a review from another team member

### How to test

1. Remove or change env vars for API to something non-valid
2. Start TruBudget with development start script
3. It should produce a correct error and don't start the app

1. Modify env vars in `api/src/envVarsSchema.ts`
2. Run `npm run generate-env-vars-docs` in api
3. Markdown file in ./api/environment-variables.md should update

Closes #1953

### CODE REVIEW

File: api/src/envVarsSchema.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+export const envVarsSchema = Joi.object({
+  ORGANIZATION: Joi.string()
+    .min(1)
+    .max(100)
+    .required()
+    .note(
+      "In the blockchain network, each node is represented by its organization name. This environment variable sets this organization name. It is used to create the organization stream on the blockchain and is also displayed in the frontend's top right corner.",
+    ),
+  PORT: Joi.number()
```
Comment (SamuelPull) [inline api/src/envVarsSchema.ts:12]: Joi.number().port()

Comment (galethil) [reply api/src/envVarsSchema.ts:16]: @SamuelPull I added the port validation in multiple variables.

File: api/src/config.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
-import logger from "./lib/logger";
-import { randomString } from "./service/hash";
+import { envVarsSchema } from "envVarsSchema";
```
Comment (SamuelPull) [inline api/src/config.ts:1]: Absolute import will fail on pipeline and probably on someone else's machine too

File: api/src/envVarsSchema.ts [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    .note(
+      "deprecated",
+      "This variable was used to choose which environment (prod or test) is used for testing the requests. The variable is deprecated now, as the Swagger documentation can be used for the prod and test environment separately.",
+    ),
+  JWT_ALGORITHM: Joi.string()
+    .default("HS256")
+    .valid("HS256", "RS256")
+    .note("Algorithm used for signing and verifying JWTs."),
+  JWT_SECRET: Joi.string()
+    .min(32)
```
Comment (SamuelPull) [inline api/src/envVarsSchema.ts:82]: Maybe 10? 😆

**Review by SamuelPull — APPROVED** [review]

**Pull request merged**